gpii/node_modules/matchMakerFramework/src/MatchMakerUtilities.js

```diff
@@ -190,25 +201,31 @@ var fluid = fluid || require("infusion"),
 
     // true if one of the values matches across the two arrays (ignoring the indexing)
     gpii.matchMakerFramework.utils.containsSameValue = function (arr1, arr2) {
-        var obj1 = {};
-        var i;
-        // convert arr1 to object for fast and easy comparison
-        for (i in arr1) {
-            obj1[arr1[i]] = true;
-        }
-        // compare:
-        for (i in arr2) {
-            if (arr2[i] in obj1) {
-                return true;
-            }
-        }
-        return false;
+        var obj2 = fluid.arrayToHash(arr2);
+        // find match across arrays or set to false
+        var found = fluid.find(arr1, function (v) {
+            return obj2[v];
+        }, false);
+        return found;
     };
 
     /**
+     * Calculate dispositions based on priorities while taking solution types into account.
+     *
+     * NOTE: this function will modify the solrecs object by adding, changing or removing the `disposition`,
+     * `dispositionReason` and `priority` properties of a solrec[solution] entry.
+     *
+     * @solrecs (Object): a map of solution information, keyed by solution ids. This object should
+     *     have a format identical to the output of the gpii.matchMakerFramework.utils.expandSolutions
+     *     function. Each entry in this object will potentially be modified by the
+     *     modification/addition/deletion of the disposition, priority and/or dispositionReason.
+     * @typeInfo (Object): contains information about the type each solution is. More precisely,
```

Comment:
Needs to be more up-front information on what a "solution type" is - I imagine this is its "apptology type"? There needs to be a central description somewhere, perhaps in a self-contained file, "apptology.md", about what these types are and how they interact with the canopy matchmaker algorithm. The GPII-404 JIRA itself is very slim and I don't think we have recorded any of this information anywhere except in some piratepads.